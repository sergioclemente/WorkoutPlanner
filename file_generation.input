(10min, 55, 75), (1hr, 80), (5min, 75, 55)
-- EOT
(10min, 55, 75), (1hr, *, Free), (5min, 75, 55)
-- EOT
(20min, 95, "Between 90-95")
-- EOT
(10min, 55, 75), 4[(1hr, 80), (5min, 55)], (20min, 75)
-- EOT
3[(1min, 90), (2min, 100), (1min, 55)]
-- EOT
2[(20s, *, MAX EFFORT), (40s, 40, easy coasting), (2min, 55)]
-- EOT
(12min, 55, 70)
8[(15s, *, Spin ups), (15s, 55)]
5[(30s, 85, 90, 95, 100, 105), (30s, 55)]
(5min, *, rest - prepare for 5s all out)
2[(5s, All out, *), (2min, *, rest)]
(5min, 55,  prepare for 20min all out)
[(20m, 104, All out), (5min, 50), 4[(15s, 120), (45s, 50)], (5min, 50)]
[(20s, *, prepare for 1min all out), (1min, *, All out), (10min, 50)]
-- EOT