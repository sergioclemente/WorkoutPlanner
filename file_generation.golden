Input: (10min, 55, 75), (1hr, 80), (5min, 75, 55)

[
    {
        "x": {
            "unit": 13,
            "value": 0,
            "estimatedDistanceInMiles": 0,
            "estimatedDurationInSeconds": 0
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "10' build from 55% to 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 10,
            "estimatedDistanceInMiles": 3.333333333333333,
            "estimatedDurationInSeconds": 600
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "10' build from 55% to 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 10,
            "estimatedDistanceInMiles": 3.333333333333333,
            "estimatedDurationInSeconds": 600
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4200,
            "estimatedDistanceInMiles": 23.333333333333332,
            "estimatedDurationInSeconds": 4200
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4200,
            "estimatedDistanceInMiles": 23.333333333333332,
            "estimatedDurationInSeconds": 4200
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "5' warm down from 75% to 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4500,
            "estimatedDistanceInMiles": 25,
            "estimatedDurationInSeconds": 4500
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' warm down from 75% to 55%",
        "tag": "if"
    }
]
----------------------------
Input: (10min, 55, 75), (1hr, *, Free), (5min, 75, 55)

[
    {
        "x": {
            "unit": 13,
            "value": 0,
            "estimatedDistanceInMiles": 0,
            "estimatedDurationInSeconds": 0
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "10' build from 55% to 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 10,
            "estimatedDistanceInMiles": 3.333333333333333,
            "estimatedDurationInSeconds": 600
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "10' build from 55% to 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 10,
            "estimatedDistanceInMiles": 3.333333333333333,
            "estimatedDurationInSeconds": 600
        },
        "y": {
            "ifValue": 0.5,
            "originalUnit": 11,
            "originalValue": 0
        },
        "label": "1hr Free",
        "tag": "free-ride"
    },
    {
        "x": {
            "unit": 12,
            "value": 4200,
            "estimatedDistanceInMiles": 18.333333333333332,
            "estimatedDurationInSeconds": 4200
        },
        "y": {
            "ifValue": 0.5,
            "originalUnit": 11,
            "originalValue": 0
        },
        "label": "1hr Free",
        "tag": "free-ride"
    },
    {
        "x": {
            "unit": 12,
            "value": 4200,
            "estimatedDistanceInMiles": 18.333333333333332,
            "estimatedDurationInSeconds": 4200
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "5' warm down from 75% to 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4500,
            "estimatedDistanceInMiles": 20,
            "estimatedDurationInSeconds": 4500
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' warm down from 75% to 55%",
        "tag": "if"
    }
]
----------------------------
Input: (20min, 95, "Between 90-95")

[
    {
        "x": {
            "unit": 13,
            "value": 0,
            "estimatedDistanceInMiles": 0,
            "estimatedDurationInSeconds": 0
        },
        "y": {
            "ifValue": 0.95,
            "originalUnit": 0,
            "originalValue": 0.95
        },
        "label": "20' \"Between 90-95\" @ 95%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 20,
            "estimatedDistanceInMiles": 6.666666666666666,
            "estimatedDurationInSeconds": 1200
        },
        "y": {
            "ifValue": 0.95,
            "originalUnit": 0,
            "originalValue": 0.95
        },
        "label": "20' \"Between 90-95\" @ 95%",
        "tag": "if"
    }
]
----------------------------
Input: (10min, 55, 75), 4[(1hr, 80), (5min, 55)], (20min, 75)

[
    {
        "x": {
            "unit": 13,
            "value": 0,
            "estimatedDistanceInMiles": 0,
            "estimatedDurationInSeconds": 0
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "10' build from 55% to 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 10,
            "estimatedDistanceInMiles": 3.333333333333333,
            "estimatedDurationInSeconds": 600
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "10' build from 55% to 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 10,
            "estimatedDistanceInMiles": 3.333333333333333,
            "estimatedDurationInSeconds": 600
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4200,
            "estimatedDistanceInMiles": 23.333333333333332,
            "estimatedDurationInSeconds": 4200
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4200,
            "estimatedDistanceInMiles": 23.333333333333332,
            "estimatedDurationInSeconds": 4200
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4500,
            "estimatedDistanceInMiles": 24.583333333333332,
            "estimatedDurationInSeconds": 4500
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 4500,
            "estimatedDistanceInMiles": 24.583333333333332,
            "estimatedDurationInSeconds": 4500
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 8100,
            "estimatedDistanceInMiles": 44.58333333333333,
            "estimatedDurationInSeconds": 8100
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 8100,
            "estimatedDistanceInMiles": 44.58333333333333,
            "estimatedDurationInSeconds": 8100
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 8400,
            "estimatedDistanceInMiles": 45.83333333333333,
            "estimatedDurationInSeconds": 8400
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 8400,
            "estimatedDistanceInMiles": 45.83333333333333,
            "estimatedDurationInSeconds": 8400
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 12000,
            "estimatedDistanceInMiles": 65.83333333333333,
            "estimatedDurationInSeconds": 12000
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 12000,
            "estimatedDistanceInMiles": 65.83333333333333,
            "estimatedDurationInSeconds": 12000
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 12300,
            "estimatedDistanceInMiles": 67.08333333333333,
            "estimatedDurationInSeconds": 12300
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 12300,
            "estimatedDistanceInMiles": 67.08333333333333,
            "estimatedDurationInSeconds": 12300
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 15900,
            "estimatedDistanceInMiles": 87.08333333333333,
            "estimatedDurationInSeconds": 15900
        },
        "y": {
            "ifValue": 0.8,
            "originalUnit": 0,
            "originalValue": 0.8
        },
        "label": "1hr @ 80%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 15900,
            "estimatedDistanceInMiles": 87.08333333333333,
            "estimatedDurationInSeconds": 15900
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 16200,
            "estimatedDistanceInMiles": 88.33333333333333,
            "estimatedDurationInSeconds": 16200
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "5' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 16200,
            "estimatedDistanceInMiles": 88.33333333333333,
            "estimatedDurationInSeconds": 16200
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "20' @ 75%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 12,
            "value": 17400,
            "estimatedDistanceInMiles": 95,
            "estimatedDurationInSeconds": 17400
        },
        "y": {
            "ifValue": 0.75,
            "originalUnit": 0,
            "originalValue": 0.75
        },
        "label": "20' @ 75%",
        "tag": "if"
    }
]
----------------------------
Input: 3[(1min, 90), (2min, 100), (1min, 55)]
-- EOT
[
    {
        "x": {
            "unit": 13,
            "value": 0,
            "estimatedDistanceInMiles": 0,
            "estimatedDurationInSeconds": 0
        },
        "y": {
            "ifValue": 0.9,
            "originalUnit": 0,
            "originalValue": 0.9
        },
        "label": "1' @ 90%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 1,
            "estimatedDistanceInMiles": 0.3333333333333333,
            "estimatedDurationInSeconds": 60
        },
        "y": {
            "ifValue": 0.9,
            "originalUnit": 0,
            "originalValue": 0.9
        },
        "label": "1' @ 90%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 1,
            "estimatedDistanceInMiles": 0.3333333333333333,
            "estimatedDurationInSeconds": 60
        },
        "y": {
            "ifValue": 1,
            "originalUnit": 0,
            "originalValue": 1
        },
        "label": "2' @ 100%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 3,
            "estimatedDistanceInMiles": 1,
            "estimatedDurationInSeconds": 180
        },
        "y": {
            "ifValue": 1,
            "originalUnit": 0,
            "originalValue": 1
        },
        "label": "2' @ 100%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 3,
            "estimatedDistanceInMiles": 1,
            "estimatedDurationInSeconds": 180
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "1' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 4,
            "estimatedDistanceInMiles": 1.25,
            "estimatedDurationInSeconds": 240
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "1' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 4,
            "estimatedDistanceInMiles": 1.25,
            "estimatedDurationInSeconds": 240
        },
        "y": {
            "ifValue": 0.9,
            "originalUnit": 0,
            "originalValue": 0.9
        },
        "label": "1' @ 90%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 5,
            "estimatedDistanceInMiles": 1.5833333333333333,
            "estimatedDurationInSeconds": 300
        },
        "y": {
            "ifValue": 0.9,
            "originalUnit": 0,
            "originalValue": 0.9
        },
        "label": "1' @ 90%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 5,
            "estimatedDistanceInMiles": 1.5833333333333333,
            "estimatedDurationInSeconds": 300
        },
        "y": {
            "ifValue": 1,
            "originalUnit": 0,
            "originalValue": 1
        },
        "label": "2' @ 100%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 7,
            "estimatedDistanceInMiles": 2.25,
            "estimatedDurationInSeconds": 420
        },
        "y": {
            "ifValue": 1,
            "originalUnit": 0,
            "originalValue": 1
        },
        "label": "2' @ 100%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 7,
            "estimatedDistanceInMiles": 2.25,
            "estimatedDurationInSeconds": 420
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "1' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 8,
            "estimatedDistanceInMiles": 2.5,
            "estimatedDurationInSeconds": 480
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "1' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 8,
            "estimatedDistanceInMiles": 2.5,
            "estimatedDurationInSeconds": 480
        },
        "y": {
            "ifValue": 0.9,
            "originalUnit": 0,
            "originalValue": 0.9
        },
        "label": "1' @ 90%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 9,
            "estimatedDistanceInMiles": 2.8333333333333335,
            "estimatedDurationInSeconds": 540
        },
        "y": {
            "ifValue": 0.9,
            "originalUnit": 0,
            "originalValue": 0.9
        },
        "label": "1' @ 90%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 9,
            "estimatedDistanceInMiles": 2.8333333333333335,
            "estimatedDurationInSeconds": 540
        },
        "y": {
            "ifValue": 1,
            "originalUnit": 0,
            "originalValue": 1
        },
        "label": "2' @ 100%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 11,
            "estimatedDistanceInMiles": 3.5,
            "estimatedDurationInSeconds": 660
        },
        "y": {
            "ifValue": 1,
            "originalUnit": 0,
            "originalValue": 1
        },
        "label": "2' @ 100%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 11,
            "estimatedDistanceInMiles": 3.5,
            "estimatedDurationInSeconds": 660
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "1' @ 55%",
        "tag": "if"
    },
    {
        "x": {
            "unit": 13,
            "value": 12,
            "estimatedDistanceInMiles": 3.75,
            "estimatedDurationInSeconds": 720
        },
        "y": {
            "ifValue": 0.55,
            "originalUnit": 0,
            "originalValue": 0.55
        },
        "label": "1' @ 55%",
        "tag": "if"
    }
]
----------------------------
